### PR TITLE
Use Modern Python 3 Syntax and Patterns

### PR DESCRIPTION
**Pull request by mattsta**
### Description

Let's use modern style for modern code.

These changes remove python 2 syntax all over the place in favor of modern python 3 syntax first released in 2008.

Commits don't introduce any functionality changes. We're just swapping legacy syntax and practices for modern commonly accepted python practices all over the place.

high profile projects using legacy syntax and styles means new people learning things copy legacy patterns without even knowing it, then they also propagate weird hybrid low-information new-legacy practices down throughout time to future codebases and articles and tutorials and books causing a cascade of the past into the future we can't ever get away from. break the cycle. do new stuff good.

(even if this is too overwhelming or too annoying to actually be merged, it's still a good reference point for new code and refactors and modern tutorial practices going forward)

### CODE REVIEW

**Comment by ezyang** [comment]
Need cla. Changes look fine but the type to isinstance changes are technically not semantics preserving and have some risk. If you split to only do super changes and turn on the lint against old style super I would be quite happy to merge.

File: caffe2/python/net_printer.py [hunk 1]
```diff
@@ -185,8 +185,8 @@ def __str__(self):
 
 class Printer(Visitor, Text):
     def __init__(self, factor_prefixes=False, c2_syntax=True):
-        super(Visitor, self).__init__()
-        super(Text, self).__init__()
+        super().__init__()
+        super().__init__()
```
Comment (kit1980) [inline caffe2/python/net_printer.py:189]: I don't think these are equivalent. In general, need to be careful with multiple inheritance.

**Review by kit1980 — CHANGES_REQUESTED** [review]
From numerous test failures looks like the code is not functionally equivalent. I really suggest to split this into multiple PRs by dir and change type, otherwise there will be constant conflicts.

**Review by malfet — CHANGES_REQUESTED** [review]
Please split this change into several PRs that address particular use of antiquated syntax.
For example: `super(XYZ, self).`->`super().` calls can be replaced with `sed` command and I suspect would be perfectly safe to land.

**Pull request closed without merging**